### PR TITLE
add handling for sparse arrays

### PR DESCRIPTION
**Pull request by toddtarsi**
Resolves https://github.com/mrjono1/joi-to-typescript/issues/324

### CODE REVIEW

**Comment by toddtarsi** [comment]
@mrjono1 - Thank you for the very impressive codebase. Please let me know if I can do anything to help expedite the process of getting this fix out, thanks!

File: src/__tests__/array/array.ts [hunk 1]
```diff
@@ -70,4 +71,29 @@ export type TestList = Test[];
  */
 export type TestList = string[];`);
   });
+
+  test('test to ensure sparse arrays have undefined as a possible type', () => {
+    // this tests this code
```
Comment (mrjono1) [inline src/__tests__/array/array.ts:76]: Can this be removed?

Comment (toddtarsi) [reply src/__tests__/array/array.ts:76]: @mrjono1 - Yep! Do you want me to remove the test and comment or just the comment?

Comment (toddtarsi) [reply src/__tests__/array/array.ts:76]: @mrjono1 - bump

Comment (mrjono1) [reply src/__tests__/array/array.ts:76]: Remove the commented out code please

File: src/parse.ts [hunk 2]
```diff
@@ -410,6 +410,8 @@ function parseStringSchema(details: StringDescribe, settings: Settings, rootSche
 
 function parseArray(details: ArrayDescribe, settings: Settings): TypeContent | undefined {
   const { interfaceOrTypeName, jsDoc } = getCommonDetails(details, settings);
+  // @ts-expect-error Idk why but the sparse flag is not in the type
+  const isSparse = details.flags?.sparse;
```
Comment (mrjono1) [inline src/parse.ts:414]: There is a custom types file for this you can add it here https://github.com/mrjono1/joi-to-typescript/blob/master/src/joiDescribeTypes.ts#L10

File: src/__tests__/array/array.ts (hunk 1 above)
Comment (toddtarsi) [reply src/__tests__/array/array.ts]: @mrjono1 - Done

File: src/parse.ts (hunk 2 above)
Comment (toddtarsi) [reply src/parse.ts:413]: @mrjono1 - Done, thanks for the tip

**Comment by toddtarsi** [comment]
@mrjono1 - All requested changes are made!

**Review by mrjono1 — APPROVED** [review]

**Comment by mrjono1** [comment]
Released, thankyou! 
https://github.com/mrjono1/joi-to-typescript/releases/tag/4.8.0

**Comment by toddtarsi** [comment]
Woohoo! Thank you @mrjono1!

**Pull request merged**